libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/nativeLoginInterface/NativeLoginManager.kt

```diff
@@ -0,0 +1,74 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.auth.nativeLoginInterface
+
+import android.content.Intent
+
+enum class NativeLoginResult {
+    InvalidUsername,      // Username does not meet Salesforce criteria (length, email format, ect).
+    InvalidPassword,      // Password does not meet the weakest Salesforce criteria.
+    InvalidCredentials,   // Username/password combination is incorrect.
+    UnknownError,
+    Success,
+}
+
+/**
+ *  Manage native login flow.
+ */
+interface NativeLoginManager {
+
+    /**
+     * If the native login view should show a back button.
+     */
+    val shouldShowBackButton: Boolean
+
+    /**
+     * The username of the locked account or null.  Can be used to pre-populate the username field
+     * or in a message telling the user which account biometric will unlock.
+     */
+    val biometricAuthenticationUsername: String?
+
+    /**
+     * Initiate a login with user provided username and password.
+     *
+     * @param username User provided Salesforce username.
+     * @param password User provided Salesforce password.
+     * @return NativeLoginResult
+     */
+    suspend fun login(username: String, password: String): NativeLoginResult
+
+    /**
+     * Use this function to get the intent for fallback web based authentication.
+     *
+     * In your native login activity start this intent in such a way as to [get a result callback](https://developer.android.com/training/basics/intents/result).
+     * A result code of `Activity.RESULT_OK` means the user has successfully authenticated via the webview and you
+     * can finish your activity.
+     *
+     * @return the intent to be started for fallback web based authentication.
+     */
+    fun getFallbackWebAuthenticationIntent() : Intent
```

Comment:
I really wanted to handle this for the customer but without an activity (or even if they give me a reference to their activity) it just isn't possible.  Good news is the code the customer will have to implement pretty simple.  Here is an example of the new (`startActivityForResult` is deprecated) way to do it:

```
private val fallbackResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
    if (result.resultCode == Activity.RESULT_OK) {
        finish()
    }
}
```

And in their button they simply call `fallbackResult.launch(nativeLoginManager.getFallbackWebAuthenticationIntent())`